lib/money/money.rb

```diff
@@ -8,6 +8,16 @@ class Money
 
   class << self
     alias_method :from_amount, :new
+
+    def new(value = 0, _currency = nil)
+      if value == 0
+        @empty ||= from_amount(0).freeze
+      else
+        from_amount(value, _currency)
+      end
+    end
```

Comment:
They are all immutable. It's just a matter of what range of values is worth caching. [Just like how the JVM caches Integer instances between -128 and 127](http://stackoverflow.com/a/3637974).

In our case, 0 is by a very far the most common value, other aren't really worth caching unless we want to do some kind of LRU, but it's taking it a bit far.